### PR TITLE
Adjusts NPC_EARTHQUAKE behavior

### LINKED ISSUE #1679: [Error]: 0 enemies targeted by 653:NPC_EARTHQUAKE, divide per 0 avoided!
Mode: Pre-Renewal
Git Hash: Latest
Bug: Core / Skill
Issue:
In the process of killing Maya Purple or Maya you will get this message:
[Error]: 0 enemies targeted by 653:NPC_EARTHQUAKE, divide per 0 avoided!

### LINKED ISSUE #3176: Earthquake don't make any damage
* **rAthena Hash**:  c68da31f67b205854ead7d10e5efb4d6c8bac19d

* **Client Date**: 20170614

* **Server Mode**: pre-renewal

* **Description of Issue**: lot of the time mvp that cast Earthquake make no damage
  * Result: when mvps ex: ifrit, rsx ect... cast earthquake player didn't get any damage
  * Expected Result: should make high damage
  * How to Reproduce: Just let a mvp cast earthquake on you
  * Official Information: I know that earthquake should do high damage.
  

* **Modifications that may affect results**: 
 I test it in a 100% clean rathena git

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: Fixes #1679 and fixes #3176

* **Server Mode**: Pre-renewal and Renewal

* **Description of Pull Request**: 
  * Converts the skill to a unit-based skill.
  * Damage is now properly calculated based on the caster's ATK.
  * Damage is considered magic with the element forced to Neutral.
  * Now ignores target's DEF and Land Protector.
  * Removed old bits of the skill.
Thanks to @poporing, @mrjnumber1, @Daegaladh, @malufett, and @exneval!

### CODE REVIEW

**Comment by Daegaladh** [comment]
Monsters seems to be damaging themselves everytime they use it

**Comment by Daegaladh** [comment]
Damage isn't being spread into all enemies in the area.
Also, Tumbling isn't being triggered by this skill.

**Comment by teededung** [comment]
Earthquake now is too strong... I don't know how monsters can deal more than 2 million damage with Earthquake

**Comment by Daegaladh** [comment]
> Earthquake now is too strong... I don't know how monsters can deal more than 2 million damage with Earthquake

Yeah, we're checking on it, the damage needs to be adjusted for both monsters and players

File: src/map/battle.cpp [hunk 1]
```diff
@@ -6170,6 +6174,14 @@ struct Damage battle_calc_magic_attack(struct block_list *src,struct block_list
 			case AB_RENOVATIO:
 				ad.damage = status_get_lv(src) * 10 + sstatus->int_;
 				break;
+			case NPC_EARTHQUAKE:
+				if (src->type == BL_PC)
+					ad.damage = sstatus->str * 2 + sstatus->rhw.atk;
+				else
+					ad.damage = sstatus->str + status_get_lv(src) + (80 + rnd() % 41) * sstatus->rhw.atk;
```
Comment (Daegaladh) [inline src/map/battle.cpp:6181]: Not entirely sure about the player damage but I think this should be something like this:

```
				if (src->type == BL_PC)
					ad.damage = sstatus->str * 2 + battle_calc_weapon_attack(src, src, skill_id, skill_lv, mflag).damage;
				else
					ad.damage = battle_calc_base_damage(src, sstatus, &sstatus->rhw, sc, tstatus->size, 0);
```

**Pull request merged**